src/lib/coda_base/pending_coinbase.ml

```diff
@@ -821,8 +843,19 @@ module T = struct
       in
       let update_stack1 stack =
         let%bind stack = update_state_stack stack in
-        let total_coinbase_amount =
-          Currency.Amount.var_of_t constraint_constants.coinbase_amount
+        let%bind total_coinbase_amount =
+          let supercharge_factor =
+            Tick0.Field.Var.constant
+              (Field.of_int constraint_constants.supercharged_coinbase_factor)
+          in
+          let coinbase_amount =
+            Currency.Amount.var_of_t constraint_constants.coinbase_amount
+          in
+          let%bind supercharged_coinbase =
+            Currency.Amount.Checked.scale supercharge_factor coinbase_amount
```

Comment:
This is technically a constant, would be good to do the multiplication on the constants and then `var_of_t`.